pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -43,11 +45,27 @@ const (
 
 var runInTestMode bool
 var runInDemoMode bool
+var maxConcurrentReconciles int 
 
 func init() {
 	flag.BoolVar(&runInTestMode, "test-mode", false, "disable ironic communication")
 	flag.BoolVar(&runInDemoMode, "demo-mode", false,
 		"use the demo provisioner to set host states")
+
+	if mcrEnv, ok := os.LookupEnv("BMO_CONCURRENCY"); ok {
+		mcr, err := strconv.Atoi(mcrEnv)
+		if err == nil && mcr > 0 {
+			log.Info(fmt.Sprintf("BMO_CONCURRENCY of %d is set via an environment variable", mcr))
+			maxConcurrentReconciles = mcr
+		} else {
+			log.Error(err, fmt.Sprintf("BMO_CONCURRENCY value: %s is invalid", mcrEnv))
+			os.Exit(1) 
+		}
```

Comment:
On the previous PR we discussed doing something more like:

```
if err != nil {
  os.Exit(1)
}
if mcr > 0 {
   maxConcurrentReconciles = mcr
}
```

so that you could pass -1 to get the default. Did you decide against that?

Reply:
+1. Makes sense this way.